components/elm/src/cpl/lnd_disagg_forc.F90

```diff
@@ -0,0 +1,797 @@
+  subroutine downscale_longwave_to_topounit(g, i, t, x2l, lnd2atm_vars, method)
+  
+    !
+    ! !DESCRIPTION:
+    ! Downscale longwave radiation from gridcell to column
+    ! Must be done AFTER temperature downscaling
+    !
+    ! !USES:
+    use clm_time_manager, only : get_nstep
+    use domainMod       , only : ldomain
+    use landunit_varcon , only : istice_mec
+    use elm_varcon      , only : lapse_glcmec
+    use elm_varctl      , only : glcmec_downscale_longwave
+    !
+    ! !ARGUMENTS:
+    integer                    , intent(in)    :: g
+    integer                    , intent(in)    :: i
+    integer                    , intent(in)    :: t
+    real(r8)                         , intent(in)    :: x2l(:,:)
+    type(lnd2atm_type)               , intent(in)    :: lnd2atm_vars
+    integer                    , intent(in)    :: method
+    !
+    ! !LOCAL VARIABLES:
+    integer  :: c,l,fc     ! indices
+    integer  :: nstep
+    real(r8) :: hsurf_t      ! column-level elevation (m)
+    real(r8) :: hsurf_g      ! gridcell-level elevation (m)
+
+    real(r8) :: tair_g         ! original gridcell mean air temperature
+    real(r8) :: tair_t         ! downscaled topounit air temperature
+    real(r8) :: tsfc_g         ! original gridcell surface temperature
+    real(r8) :: tsfc_t         ! downscaled topounit surface temperature
+    real(r8) :: lwrad_g        ! original gridcell mean LW radiation
+    real(r8) :: lwrad_t        ! downscaled topounit LW radiation
+    real(r8) :: newsum_lwrad_g ! weighted sum of column-level lwrad after normalization
+
+    character(len=*), parameter :: subname = 'downscale_longwave_top'
+    !-----------------------------------------------------------------------
+
+    nstep = get_nstep()
+    
+    ! Do the downscaling
+    hsurf_g = grc_pp%elevation(g) !ldomain%topo2(g)
+    hsurf_t = top_pp%elevation(t)
+
+    ! Here we assume that deltaLW = (dLW/dT)*(dT/dz)*deltaz
+    ! We get dLW/dT = 4*eps*sigma*T^3 = 4*LW/T from the Stefan-Boltzmann law,
+    ! evaluated at the mean temp.
+    ! We assume the same temperature lapse rate as above.
+
+    tair_g = x2l(index_x2l_Sa_tbot,i)
+    tair_t = top_as%tbot(t)
+    tsfc_g = lnd2atm_vars%t_rad_grc(g)
+    lwrad_g = x2l(index_x2l_Faxa_lwdn,i)
+    if (method == 0 .or. (method == 1 .and. nstep == 0)) then
+       lwrad_t = lwrad_g - &
+          4.0_r8 * lwrad_g/(0.5_r8*(tair_t+tair_g)) * &
+          lapse_glcmec * (hsurf_t - hsurf_g)
+    else
+       
+       lwrad_t = lwrad_g + lnd2atm_vars%eflx_lwrad_out_grc(g) * &
+          4._r8 * (tair_t - tair_g) / tsfc_g
+       
+    end if
+       top_af%lwrad(t) = lwrad_t
+
+  end subroutine downscale_longwave_to_topounit
+  
+  !-------------------------------------------------------
+  ! Downscale precipitation from grid to topounit using FNM method
+  subroutine downscale_precip_to_topounit_FNM(mxElv,uovern_t,grdElv,topoElv,rain_g,snow_g,deltaR,deltaS,hrise) 
```

Comment:
Add a comment here to explain what FNM method is

Reply:
@thorntonpe I have added comment describing FNM method in my latest commit. Thanks.